pkg/utils/names/names.go

```diff
@@ -72,3 +73,7 @@ func RandomName(length int, chars string) string {
 	}
 	return string(randomName)
 }
+
+func ForStack(name string) string {
+	return strings.ReplaceAll(name, "_", "-")
+}
```

Comment:
I am not a fan of this. We don't manipulate user input for cluster or nodegroup names. This might have undesired effects on the user.

Reply:
Yeah, I am also not a fan. I'm trying to open the discussion for a solution on this problem. Not sure if this is the reason why you didn't support underscore in the first place?

Reply:
We didn't support underscore because it wasn't supported in CFN before. This validation was added a long time ago. See here for more details: https://github.com/weaveworks/eksctl/issues/2943
https://github.com/weaveworks/eksctl/issues/2943#issuecomment-859390934

Reply:
I suspected that was the case. Thanks for the link!